lib/opam_solve.ml

```diff
@@ -95,11 +95,12 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
       Opam.depends_on_dune ~allow_jbuilder depends
       || Opam.depends_on_dune ~allow_jbuilder depopts
     in
+    let has_no_build_command = List.is_empty (OpamFile.OPAM.build opam_file) in
     let summary = Opam.Package_summary.from_opam pkg opam_file in
     let is_valid_dune_wise =
       Opam.Package_summary.is_base_package summary
       || Opam.Package_summary.is_virtual summary
```

Comment:
I'm wondering, wouldn't it make sense to add this to `is_virtual`? Or a new category in `Opam.Package_summary`?

Reply:
I considered that and conceptually it would make sense for packages without build commands to be considered "virtual". We could add a new field to `Package_summary.t`:
```ocaml
build : OpamTypes.command list
```
The problem is that in order to update `Package_summary.equal` and `Package_summary.pp` we'd need to implement equality and pretty-printing for `OpamTypes.command` and it looks like that would add a lot of code.

However, it appears that neither `Package_summary.equal` nor 'Package_summary.pp' are ever called, so perhaps they can just be deleted.

Reply:
I've updated this PR so that the check is now done inside `is_virtual`